lib/puppet/parser/functions/size.rb

```diff
@@ -37,6 +37,8 @@ module Puppet::Parser::Functions
 
     elsif item.is_a?(Array)
       result = item.size
+    elsif item.is_a?(Hash)
```

Comment:
you can put this as `|| item.is_a?(Hash)` at the end of L38 and save two lines here.